src/frontend/Symbol_table.ml

```diff
@@ -5,75 +5,71 @@ open Core_kernel
 (* TODO: I'm sure this implementation could be made more efficient if that's necessary. There's no need for all the string comparison.
 We could just keep track of the count of the entry into the hash table and use that for comparison. *)
 type 'a state =
-  { table: (string, 'a) Hashtbl.t
-  ; stack: string Stack.t
-  ; scopedepth: int ref
-  ; readonly: (string, unit) Hashtbl.t
-  ; isunassigned: (string, unit) Hashtbl.t
-  ; globals: (string, unit) Hashtbl.t }
+  { table: (string, 'a, String.comparator_witness) Map.t
+  ; stack: string List.t
+  ; scopedepth: int
+  ; readonly: (string, unit, String.comparator_witness) Map.t
+  ; isunassigned: (string, unit, String.comparator_witness) Map.t
+  ; globals: (string, unit, String.comparator_witness) Map.t }
```

Comment:
`Map`s from `'a` to `unit` are are the same as `Set`s of `'a`. I was using this as a way of getting a Hashset which Core_kernel seemed to not supply.